src/guide_restli_server.md

```diff
@@ -1755,6 +1767,10 @@ Valid parameter types and return types for action are:
     example, `StringMap`, `LongMap`, and so on.
 -   Custom types
 
+To use Arrays and Maps of non-primitive type (Eg record, enum) as return type,
+the array/map should be [defined as data models](#defining-data-models). This is needed for the
+template classes to be generated for them. Eg, `typeref FooRecordArrayRef = array[FooRecord]`
```

Comment:
By "template classes" do you mean "java-binding classes"?  Should we elaborate this? Actually in this doc, the word "template" almost always refers to "resource templates" https://linkedin.github.io/rest.li/user_guide/restli_server

Reply:
nit: "E.g." might be more formal than "Eg"